Sources/UIView.swift

```diff
@@ -206,7 +206,13 @@ open class UIView: UIResponder, CALayerDelegate, UIAccessibilityIdentification {
     /// Adds a subview without touching the view's layer or any of its sublayers.
     /// We need to be able to add layers at an index that isn't directly related to its subview index.
     private func insertSubviewWithoutTouchingLayer(_ view: UIView, at index: Int) {
-        if view.superview != nil { view.removeFromSuperview() }
+        // remove from superview without removing from superlayer
+        if let oldSuperview = view.superview {
+            oldSuperview.subviews = oldSuperview.subviews.filter { $0 != view }
+            view.superview = nil
+            oldSuperview.setNeedsLayout()
+        }
```

Comment:
pretty much what happens in `removeFromSuperview`, but without `layer.removeFromSuperlayer()`

Reply:
I don't fully get why we should not touch the player in this case. Can you explain that @rikner ?

Reply:
Do mean the player or the layer? 

So we don't want to touch the **layer** here since this function is called this way ^^ (`insertSubviewWithoutTouchingLayer`). I probably wouldn't have found the bug if not for the function name.

In case you mean **player**... not sure if we could fix it there, but I think fixing it here makes more sense in order to get more close to the iOS behavior.